Sources/SwiftSyntax/Documentation.docc/Working with SwiftSyntax.md

```diff
@@ -0,0 +1,221 @@
+the Swift parser, swift-format, and Swift macros are built.
+
+## Syntax Trees
+
+SwiftSyntax is a library for representing Swift source code with a data structure
+called a *syntax tree*. The structure of a syntax tree reflects not just the
+relative order and position of text and whitespace in a file 
+\- its "lexical structure" - but also the relationship between nested syntactic
+elements and their neighbors - its "syntactic structure".
+
+The syntax tree was designed with three tenets in mind:
+
+### Immutability
+
+The syntax tree is a fully immutable [persistent data structure][PersistentDataStructure]. 
+Once created, the syntax tree never changes, and may only be modified by
+creating a new syntax tree. The SwiftSyntax library provides a number of APIs 
+for modifying an existing syntax tree efficiently and safely. It does so by
+copying only the minimal amount of structure that has changed into the new
+tree. Any unchanged structure from the original tree becomes part of a shared 
+core that is transparently wrapped by the new syntax tree. In this way, the
+SwiftSyntax library reuses and shares as much data as possible during syntactic 
+transformations. This reduces the overall memory footprint and the overhead
+normally associated with copying large and complex data structures like 
+syntax trees.
+
+Immutability also comes with the benefits of fearless concurrency. It is always
+safe to manipulate a syntax tree across threads. Two threads mutating the same
+tree will recieve two independent copies of their respective modified syntax
+trees that will never interfere with each other. Moreover, modifications to 
+the syntax tree do not involve locks or mutexes.
+
+### Source Fidelity
+
+The SwiftSyntax library is designed to represent not just text, but 
+whitespace, comments, compiler directives, and otherwise invisible data like 
+the [Unicode byte order mark][BOM]. Collectively, these non-syntax entities are
+referred to as "trivia". The syntax tree holds on to every byte of
+information in the source text, and can render the original document
+with full accuracy and precision. This property is referred to as 
+"source fidelity", because the syntax tree is always a faithful recreation of 
+the source code it was given as input.
+
+### Resilience
+
+To acchieve source fidelity, both well-formed and ill-formed source code must
+be equally representable in the syntax tree. The syntax tree can represent two
+broad classes of errors: missing syntax and unexpected syntax. Missing syntax
+provides a spot for a syntax element that was expected by a parser or lexical
+analyzer but was not present in source.
+
+```
+func (uhoh: Void) {}
+     ^ Missing identifier
+```
+
+Missing syntactic elements are represented in the tree as nodes with a 
+``SourcePresence/missing`` source presence. These missing nodes are just as
+much a part of the syntax tree as ``SourcePresence/present`` nodes, and
```

Comment:
Since only tokens have presence, we can change this to 
```suggestion
``SourcePresence/missing`` source presence. These missing tokens are just as
much a part of the syntax tree as ``SourcePresence/present`` tokens, and
```